File: PHP/FilmBioskop.php
<?php
require_once __DIR__ . '/Film.php';

class FilmBioskop extends Film {
    protected $studio;
    protected $hargaTiket;
    protected $jadwalTayang;
    protected $gambar;

    // Konstruktor ini membuat objek FilmBioskop dengan informasi penayangan di bioskop.
    public function __construct($id = 0, $judul = "", $genre = "", $durasi = 0, $sutradara = "", $tahunRilis = 0, $studio = "", $hargaTiket = 0, $jadwalTayang = "", $gambar = "") {
        parent::__construct($id, $judul, $genre, $durasi, $sutradara, $tahunRilis);
        $this->studio = $studio;
        $this->hargaTiket = (double)$hargaTiket;
        $this->jadwalTayang = $jadwalTayang;
        $this->gambar = $gambar;
    }

    // Method ini mengembalikan nama studio pemutaran.
    public function getStudio(){
        return $this->studio; 
    }

    // Method ini mengubah nama studio pemutaran.
    public function setStudio($studio){
        $this->studio = $studio; 
    }

    // Method ini mengembalikan harga tiket film.
    public function getHargaTiket(){
        return $this->hargaTiket; 
    }

    // Method ini mengubah harga tiket film.
    public function setHargaTiket($hargaTiket){
        $this->hargaTiket = (double)$hargaTiket; 
    }

    // Method ini mengembalikan jadwal tayang film.
    public function getJadwalTayang(){
        return $this->jadwalTayang; 
    }

    // Method ini mengubah jadwal tayang film.
    public function setJadwalTayang($jadwalTayang){
        $this->jadwalTayang = $jadwalTayang; 
    }

    // Method ini mengembalikan lokasi poster film.
    public function getGambar(){
        return $this->gambar; 
    }

    // Method ini mengubah lokasi poster film.
    public function setGambar($gambar){
        $this->gambar = $gambar; 
    }

    // Method ini menampilkan seluruh data film beserta informasi bioskopnya.
    public function tampilkanData(){
        parent::tampilkanData();
        echo "Studio : " . $this->studio . "<br>";
        echo "Harga Tiket : Rp " . $this->hargaTiket . "<br>";
        echo "Jadwal Tayang : " . $this->jadwalTayang . "<br>";
        echo "Gambar : " . $this->gambar . "<br>";
    }
}
?>

File: Python/FilmBioskop.py
from Film import Film

class FilmBioskop(Film):
    # Konstruktor ini membuat objek FilmBioskop dengan informasi penayangan di bioskop.
    def __init__(self, id=0, judul="", genre="", durasi=0, sutradara="", tahun_rilis=0, studio="", harga_tiket=0, jadwal_tayang=""):
        super().__init__(id, judul, genre, durasi, sutradara, tahun_rilis)
        self.__studio = studio
        self.__harga_tiket = harga_tiket
        self.__jadwal_tayang = jadwal_tayang

    # Method ini mengembalikan nama studio pemutaran.
    def getStudio(self):
        return self.__studio

    # Method ini mengubah nama studio pemutaran.
    def setStudio(self, studio):
        self.__studio = studio

    # Method ini mengembalikan harga tiket film.
    def getHargaTiket(self):
        return self.__harga_tiket

    # Method ini mengubah harga tiket film.
    def setHargaTiket(self, harga_tiket):
        self.__harga_tiket = harga_tiket

    # Method ini mengembalikan jadwal tayang film.
    def getJadwalTayang(self):
        return self.__jadwal_tayang

    # Method ini mengubah jadwal tayang film.
    def setJadwalTayang(self, jadwal_tayang):
        self.__jadwal_tayang = jadwal_tayang

    # Method ini menampilkan seluruh data film beserta informasi bioskopnya.
    def tampilkanData(self):
        super().tampilkanData()
        print(f"Studio : {self.__studio}")
        print(f"Harga Tiket : Rp {self.__harga_tiket}")
        print(f"Jadwal Tayang : {self.__jadwal_tayang}")
